src/relayer/handles.test.ts

```diff
@@ -31,37 +33,42 @@ const DATA: {
   handles: [
     '0x528d1c1ab80bf0f82867971f56b3345aecbbe45d61000000000000aa36a70400',
   ],
+  userAddress: '0x37AC010c1c566696326813b840319B58Bb5840E4',
+  contractAddress: '0xb2a8A265dD5A27026693Aa6cE87Fb21Ac197b6b9',
 };
 
 describe('handles', () => {
   it('compute from ZkPok', () => {
-    const h1 = FhevmHandle.fromZKProof({
+    const zkProof = ZKProof.fromComponents({
       ciphertextWithZKProof: hexToBytes(DATA.ciphertextWithZKProof),
-      aclAddress: DATA.aclAddress as ChecksummedAddress,
-      chainId: DATA.chainId,
-      fheTypeIds: [fheTypeIdFromEncryptionBits(DATA.bitwidths[0])],
-      ciphertextVersion: DATA.ciphertextVersion,
+      aclContractAddress: DATA.aclAddress as ChecksummedAddress,
+      chainId: BigInt(DATA.chainId),
+      encryptionBits: DATA.bitwidths,
+      userAddress: DATA.userAddress,
+      contractAddress: DATA.contractAddress,
     });
+    const h1 = FhevmHandle.fromZKProof(zkProof, DATA.ciphertextVersion);
     expect(h1[0].toBytes32Hex()).toEqual(DATA.handles[0]);
   });
 
   it('equivalence testing', () => {
-    const args = {
+    const ciphertextVersion = 123;
+    const zkProof = ZKProof.fromComponents({
       ciphertextWithZKProof: hexToBytes('0xdeadbeef'),
-      aclAddress: '0xBBC1fFCdc7C316aAAd72E807D9b0272BE8F84DA0' as `0x${string}`,
-      chainId: 12345,
-      fheTypeIds: [8 as FheTypeId],
-      bitWidths: [256],
-      ciphertextVersion: 213,
-      index: 0,
-    };
-    const h1 = FhevmHandle.fromZKProof(args);
+      aclContractAddress:
+        '0xBBC1fFCdc7C316aAAd72E807D9b0272BE8F84DA0' as `0x${string}`,
+      chainId: BigInt(12345),
```

Comment:
Using the `123n` convention rather than the `BigInt(123)` constructor provides better readability imo 
```suggestion
      chaindId: 12345n
```

Reply:
fixed!